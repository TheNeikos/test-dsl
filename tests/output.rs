#[test]
fn check_invalid() {
    let ts = test_dsl::TestDsl::<()>::new();

    let tc = ts.parse_document(
        r#"
            testcase {
            }
            tetcase {
            }
            foobar {
            }
            asd
        "#,
    );

    insta::assert_snapshot!(format!("{:?}", miette::Error::new(tc.unwrap_err())));
}

#[test]
fn check_unknown() {
    let ts = test_dsl::TestDsl::<()>::new();

    let tc = ts.parse_document(
        r#"
            testcase {
                repeat 2 {
                    not_found
                }
            }
        "#,
    );

    insta::assert_snapshot!(format!("{:?}", miette::Error::new(tc.unwrap_err())));
}

#[test]
fn check_missing_argument() {
    let ts = test_dsl::TestDsl::<()>::new();

    let tc = ts.parse_document(
        r#"
            testcase {
                repeat {
                }
            }
        "#,
    );

    insta::assert_snapshot!(format!("{:?}", miette::Error::new(tc.unwrap_err())));
}

#[test]
fn check_argument_wrong_type() {
    let ts = test_dsl::TestDsl::<()>::new();

    let tc = ts.parse_document(
        r#"
            testcase {
                repeat hello {
                }
            }
        "#,
    );

    insta::assert_snapshot!(format!("{:?}", miette::Error::new(tc.unwrap_err())));
}
