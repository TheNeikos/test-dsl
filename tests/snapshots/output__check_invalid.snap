---
source: tests/output.rs
expression: "format!(\"{:?}\", miette::Error::new(tc.unwrap_err()))"
---
  [31mÃ—[0m An error occurred while parsing testcases

Error: 
  [31mÃ—[0m Not a valid test case
   â•­â”€[[36;1;4mtest.kdl:4:13[0m]
 [2m3[0m â”‚             }
 [2m4[0m â”‚             tetcase {
   Â· [35;1m            â”€â”€â”€â”¬â”€â”€â”€[0m
   Â·                [35;1mâ•°â”€â”€ [35;1mExpected a `testcase`[0m[0m
 [2m5[0m â”‚             }
   â•°â”€â”€â”€â”€
[36m  help: [0mThe outer items must all be `testcase`s

Error: 
  [31mÃ—[0m Not a valid test case
   â•­â”€[[36;1;4mtest.kdl:6:13[0m]
 [2m5[0m â”‚             }
 [2m6[0m â”‚             foobar {
   Â· [35;1m            â”€â”€â”€â”¬â”€â”€[0m
   Â·                [35;1mâ•°â”€â”€ [35;1mExpected a `testcase`[0m[0m
 [2m7[0m â”‚             }
   â•°â”€â”€â”€â”€
[36m  help: [0mThe outer items must all be `testcase`s

Error: 
  [31mÃ—[0m Not a valid test case
   â•­â”€[[36;1;4mtest.kdl:8:13[0m]
 [2m7[0m â”‚             }
 [2m8[0m â”‚             asd
   Â· [35;1m            â”€â”¬â”€[0m
   Â·              [35;1mâ•°â”€â”€ [35;1mExpected a `testcase`[0m[0m
 [2m9[0m â”‚         
   â•°â”€â”€â”€â”€
[36m  help: [0mThe outer items must all be `testcase`s
